interceptor/interceptor2.go

```diff
@@ -0,0 +1,41 @@
+package interceptor
+
+import (
+	"context"
+	"io"
+
+	"github.com/pion/rtcp"
+	"github.com/pion/rtp"
+	"github.com/pion/webrtc/v3"
+)
+
+type Interceptor2 interface {
```

Comment:
I am all for going with `Interceptor2`! What ever you feel most passionate about, I support your design.

With design stuff I like doing the 'surgical team' route. I support the decisions you make. I will prototype/test/implement the code though! I just think things are better when one person really owns it
and is passionate :)

Reply:
updated transportcc example, looks much cleaner now (no more magic for linking the different readers/writers together)